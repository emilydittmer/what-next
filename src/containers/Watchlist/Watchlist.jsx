import React, { Component } from 'react';
import '../SuggestionContainer/SuggestionContainer.scss';
import { connect } from 'react-redux';
import SuggestionCard from '../SuggestionCard/SuggestionCard'

class Watchlist extends Component {
  constructor() {
    super();
  }

  showWatchlist() {
    if (this.props.watchlist.length === 0) {
      return(<h3>Please add shows to your watchlist.</h3>)
    } else {
      return this.props.watchlist.map(show => (
        <SuggestionCard {...show} key={show.id} />
      ))
    }
  }
  render() {
    return(
      <section>
        <h2>Your Watchlist</h2>
        <div className='suggestion-container'>
          {this.showWatchlist()}
        </div>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  watchlist: state.watchlist
})


export default connect(mapStateToProps)(Watchlist);