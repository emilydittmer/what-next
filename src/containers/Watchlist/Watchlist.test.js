import React from 'react';
import Watchlist from './Watchlist';
import { shallow } from 'enzyme';

describe('Watchlist', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      watchlist: {watchlist: [{id: 1, name:'The Office'}]}
    },
    wrapper = shallow(<Watchlist {...props}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  
})