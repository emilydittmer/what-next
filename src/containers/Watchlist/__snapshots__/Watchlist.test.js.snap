// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watchlist should match the snapshot 1`] = `ShallowWrapper {}`;
